src/timer.rs

```diff
@@ -0,0 +1,60 @@
+use std::{thread, time};
+use bus::Bus;
+use std::io::{self, Write};
+
+const SAVE: &str = "\x1b7";
+const RESTORE: &str = "\x1b8";
+const MOVE_TO_TOP_LEFT: &str = "\x1b[1;1H";
+const RED: &str = "\x1b[1;31m";
+const ERASE_LINE: &str = "\x1b[2K";
+const RESET: &str = "\x1b[0m";
+
+const TOTAL_TRANSACTION_TIME: u64 = 30;
+const REFRESH_INTERVAL: u64 = 100;
+
+pub struct Timer {
+    bus: Bus<String>
```

Comment:
I think we should try do this without a new dependency. Each new dependency needs to carry its weight. So, we're gaining a new feature but is it that much better than what the stdlib or tokio offer? For example, tokio already has synchronization primitives (https://docs.rs/tokio/0.2.3/tokio/sync/index.html) that might be good enough here.

Adding dependencies is not evil, but they do increase compile times (more to link, fresh builds are slower) and require maintenance as new versions come out or there are security issues (maybe in them, or what they drag in too).

Reply:
I agree with the concern about adding new dependencies. However, `bus` provides a zero buffer broadcast function that is unavailable in `tokio` (the most similar function would be [watch](https://docs.rs/tokio/0.2.3/tokio/sync/watch/index.html), but this stores the latest value, which results in a bug when `stop_timer` is called and there are zero subscribers to the bus, then `start_timer` is called)

Reply:
What about notify? Or, if we use a task, we could cancel it?